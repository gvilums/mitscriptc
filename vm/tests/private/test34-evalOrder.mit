x = 0;
foo = fun() {
    global x;
    x = x + 1;
    return x;
};
bar = fun() {
    global x;
    x = x - 1;
    return x;
};
print("Should be i1i2: " + (("i" + foo()) + ("i" + foo())));
print("Should be 5: " + (foo() + bar()));
print("Should be 6: " + (foo() * bar()));
print("Should be false: " + (foo() > foo()));
print("Should be false: " + (foo() >= foo()));
print("Should be true: " + (foo() < foo()));
print("Should be true: " + (foo() <= foo()));
print("Should be -1: " + (foo() - foo()));
print("Should be 0: " + (foo() / foo()));
