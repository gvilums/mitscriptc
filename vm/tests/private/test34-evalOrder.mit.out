Should be i1i2: i1i2
Should be 5: 5
Should be 6: 6
Should be false: false
Should be false: false
Should be true: true
Should be true: true
Should be -1: -1
Should be 0: 0
